app/src/main/java/com/github/se/icebreakrr/ui/profile/ProfileEdit.kt

```diff
@@ -1,215 +1,229 @@
+
+                // Catchphrase Input
+                OutlinedTextField(
+                    value = catchphrase,
+                    onValueChange = { catchphrase = it },
+                    label = {
+                      Text("Catchphrase", modifier = Modifier.testTag("catchphraseLabel"))
                     },
-                    modifier = Modifier.testTag("alertDialog"))
+                    textStyle = TextStyle(fontSize = textSize.value.sp * 0.6),
+                    modifier =
+                        Modifier.fillMaxWidth().height(catchphraseHeight).testTag("catchphrase"))
+
+                Spacer(modifier = Modifier.height(padding))
+
+                // Description Input
+                OutlinedTextField(
+                    value = description,
+                    onValueChange = { description = it },
+                    label = { Text("Description") },
+                    textStyle = TextStyle(fontSize = textSize.value.sp * 0.6),
+                    modifier =
+                        Modifier.fillMaxWidth().height(descriptionHeight).testTag("description"))
+                Spacer(modifier = Modifier.height(padding))
+
+                TagSelector(
+                    selectedTag =
+                        selectedTags.map { tag -> Pair(tag, tagsViewModel.tagToColor(tag)) },
+                    outputTag =
+                        tagsSuggestions.value.map { tag ->
+                          Pair(tag, tagsViewModel.tagToColor(tag))
+                        },
+                    stringQuery = stringQuery.value,
+                    expanded = expanded,
+                    onTagClick = { tag -> tagsViewModel.removeFilter(tag) },
+                    onStringChanged = { tagsViewModel.setQuery(it, selectedTags) },
+                    textColor = MaterialTheme.colorScheme.onSurface,
+                    onDropDownItemClicked = { tag -> tagsViewModel.addFilter(tag) },
+                    height = screenHeight,
+                    width = screenWidth,
+                    textSize = (screenHeight.value * 0.03).sp,
+                )
+
+                Spacer(modifier = Modifier.height(padding))
+
+                // Modal for unsaved changes on leave page with back button
+                if (showDialogBack) {
+                  AlertDialog(
+                      onDismissRequest = { showDialogBack = false },
+                      title = { Text("You are about to leave this page") },
+                      text = { Text("Your changes will not be saved.") },
+                      confirmButton = {
+                        TextButton(
+                            onClick = {
+                              showDialogBack = false
+                              tagsViewModel.leaveUI()
+                              navigationActions.goBack()
+                            }) {
+                              Text("Discard changes")
+                            }
+                      },
```

Comment:
nitpick: could you add a boolean variable or something to check if something was changed and only then show the prompt when going back? Otherwise the prompt adds friction for nothing.

Reply:
I actually thought about doing it but I think something blocked me. I will try again.